compiler/src/dotty/tools/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -799,20 +811,16 @@ trait BCodeSkelBuilder extends BCodeHelpers {
 
         def emitNormalMethodBody(): Unit = {
           val veryFirstProgramPoint = currProgramPoint()
-          genLoad(trimmedRhs, returnType)
-
-          trimmedRhs match {
-            case (_: Return) | Block(_, (_: Return)) => ()
-            case (_: Apply) | Block(_, (_: Apply)) if trimmedRhs.symbol eq defn.throwMethod => ()
```

Comment:
These two optimizations are also taken care of by the general `LoadDestination` infrastructure.